File: api/src/main/java/io/github/mh321productions/serverapi/util/permission/DefaultRank.java
package io.github.mh321productions.serverapi.util.permission;

/**
 * Eine Sammlung der Ränge auf unserem Server
 * @author 321Productions
 *
 */
public enum DefaultRank {
	Default("default"),
	
	Premium("premium"),
	
	YouTube("youtube"),
	
	Creator("creator"),
	
	VIP("vip"),
	
	Team("team"),
	
	BuilderTest("buildertest"),
	
	Builder("builder"),
	
	BuilderPlus("builder+"),
	
	DevTest("devtest"),
	
	Dev("dev"),
	
	DevPlus("dev+"),
	
	Sup("sup"),
	
	SupPlus("sup+"),
	
	Mod("mod"),
	
	ModSr("modSr"),
	
	Admin("admin"),
	
	Nicked("nicked");
	
	
	public final String name;
	
	private DefaultRank(String internalName) {
		name = internalName;
	}
	
	public static DefaultRank fromRank(Rank rank) {
		for (DefaultRank r: values()) if (rank.equals(r)) return r;
		
		return null;
	}
}

File: src/main/kotlin/io/github/mh321productions/serverapi/util/permission/DefaultRank.kt
package io.github.mh321productions.serverapi.util.permission

/**
 * Eine Sammlung der Ränge auf unserem Server
 * @author 321Productions
 */
enum class DefaultRank(val rankName: String) {
    Default("default"),

    Premium("premium"),

    YouTube("youtube"),

    Creator("creator"),

    VIP("vip"),

    Team("team"),

    BuilderTest("buildertest"),

    Builder("builder"),

    BuilderPlus("builder+"),

    DevTest("devtest"),

    Dev("dev"),

    DevPlus("dev+"),

    Sup("sup"),

    SupPlus("sup+"),

    Mod("mod"),

    ModPlus("modSr"),

    Admin("admin"),

    Nicked("nicked");

    companion object {
        fun fromRank(rank: Rank): DefaultRank? = entries.find { it.rankName == rank.name }
    }
}
